### PR TITLE
Refactor `src`

### PR DESCRIPTION
**Pull request by jlvandenhout**
# Description of change

This refactors the `src` directory to have a more logical separation between the different environments.

## Type of change

- Enhancement (a non-breaking change which adds functionality)

## Change checklist

Tick the boxes that are relevant to your changes, and delete any items that are not.

- [x] I have followed the [contribution guidelines](https://github.com/iota-wiki/iota-wiki/blob/main/.github/CONTRIBUTING.md) for this project
- [x] I have performed a self-review of my own changes
- [x] I have made sure that added/changed links still work
- [ ] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

File: src/iota/components/HomeLayout/AppLibrariesSection.tsx [hunk 1]
```diff
@@ -6,7 +6,7 @@ import {
   Streams,
   Stronghold,
   Wallet,
-} from '../../../icons';
+} from '../../../common/icons';
```
Comment (Dr-Electron) [inline src/iota/components/HomeLayout/AppLibrariesSection.tsx:9]: Any reason why we not also use `@site` here?
```suggestion
} from '@site/src/common/icons';
```

**Review by Dr-Electron — APPROVED** [review]
LGTM. Except the @site thingy. Why do we use it in some cases and in some we don't?

File: src/iota/components/HomeLayout/AppLibrariesSection.tsx (hunk 1 above)
Comment (jlvandenhout) [reply src/iota/components/HomeLayout/AppLibrariesSection.tsx:9]: You are right, I missed a few, I'll check again.

File: src/iota/components/HomeLayout/CoreLibrariesSection.tsx [hunk 2]
```diff
@@ -6,7 +6,7 @@ import {
   Streams,
   Stronghold,
   Wallet,
-} from '../../../icons';
+} from '../../../common/icons';
```
Comment (Dr-Electron) [inline src/iota/components/HomeLayout/CoreLibrariesSection.tsx:9]:
```suggestion
} from '@site/src/common/icons';
```

File: src/iota/components/HomeLayout/index.tsx [hunk 3]
```diff
@@ -18,7 +18,7 @@ import {
   Tips,
   Tutorials,
   Wasp,
-} from '../../../icons';
+} from '../../../common/icons';
```
Comment (Dr-Electron) [inline src/iota/components/HomeLayout/index.tsx:21]:
```suggestion
} from '@site/src/common/icons';
```

File: src/utils/socials.ts [hunk 4]
```diff
@@ -7,7 +7,7 @@ import {
   LinkedIn,
   Instagram,
   Facebook,
-} from '../icons';
+} from '../common/icons';
```
Comment (Dr-Electron) [inline src/utils/socials.ts:10]:
```suggestion
} from '@site/src/common/icons';
```

File: src/shimmer/css/custom.css [hunk 5]
```diff
@@ -1,5 +1,5 @@
-@import '../fonts.css';
-@import '../patches.css';
+@import '../../common/css/fonts.css';
+@import '../../common/css/patches.css';
 @import 'variables.css';
```
Comment (Dr-Electron) [inline src/shimmer/css/custom.css:3]:
```suggestion
@import '@site/src/common/css/fonts.css';
@import '@site/src/common/css/patches.css';
@import 'variables.css';
```

File: src/shimmer/components/HomeLayout/index.tsx [hunk 6]
```diff
@@ -16,7 +16,7 @@ import {
   Tips,
   Tutorials,
   Wasp,
-} from '../../../icons';
+} from '../../../common/icons';
```
Comment (Dr-Electron) [inline src/shimmer/components/HomeLayout/index.tsx:19]:
```suggestion
} from '@site/src/common/icons';
```

File: src/shimmer/components/HomeLayout/CoreLibrariesSection.tsx [hunk 7]
```diff
@@ -1,6 +1,6 @@
 import React, { FC } from 'react';
 import Link from '@docusaurus/Link';
-import { Shimmer, Wallet } from '../../../icons';
+import { Shimmer, Wallet } from '../../../common/icons';
```
Comment (Dr-Electron) [inline src/shimmer/components/HomeLayout/CoreLibrariesSection.tsx:3]:
```suggestion
import { Shimmer, Wallet } from '@site/src/common/icons';
```

File: src/shimmer/components/HomeLayout/AppLibrariesSection.tsx [hunk 8]
```diff
@@ -1,6 +1,6 @@
 import React, { FC } from 'react';
 import Link from '@docusaurus/Link';
-import { Identity, SmartContracts, Stronghold } from '../../../icons';
+import { Identity, SmartContracts, Stronghold } from '../../../common/icons';
```
Comment (Dr-Electron) [inline src/shimmer/components/HomeLayout/AppLibrariesSection.tsx:3]:
```suggestion
import { Identity, SmartContracts, Stronghold } from '@site/src/common/icons';
```

File: src/next/css/custom.css [hunk 9]
```diff
@@ -1,5 +1,5 @@
-@import '../fonts.css';
-@import '../patches.css';
+@import '../../common/css/fonts.css';
+@import '../../common/css/patches.css';
```
Comment (Dr-Electron) [inline src/next/css/custom.css:2]:
```suggestion
@import '@site/src/common/css/fonts.css';
@import '@site/src/common/css/patches.css';
```

File: src/next/components/HomeLayout/index.tsx [hunk 10]
```diff
@@ -16,7 +16,7 @@ import {
   Tips,
   Tutorials,
   Wasp,
-} from '../../../icons';
+} from '../../../common/icons';
```
Comment (Dr-Electron) [inline src/next/components/HomeLayout/index.tsx:19]:
```suggestion
} from '@site/src/common/icons';
```

File: src/next/components/HomeLayout/CoreLibrariesSection.tsx [hunk 11]
```diff
@@ -1,6 +1,6 @@
 import React, { FC } from 'react';
 import Link from '@docusaurus/Link';
-import { Shimmer, Wallet } from '../../../icons';
+import { Shimmer, Wallet } from '../../../common/icons';
```
Comment (Dr-Electron) [inline src/next/components/HomeLayout/CoreLibrariesSection.tsx:3]:
```suggestion
import { Shimmer, Wallet } from '@site/src/common/icons';
```

**Review by Dr-Electron — CHANGES_REQUESTED** [review]

**Review by Dr-Electron — APPROVED** [review]

**Pull request merged**